_applications/bad-apples.adoc

```diff
@@ -0,0 +1,310 @@
+that periodically reads from the database to display the number of detected
+fraudlent transactions. Using this service you can easily monitor how a change
+to the filter job affects the number of detected transactions.
+
+=== Record selection process
+
+The filter job service will examine all the records in the database and use
+a few simple algorithms to select the fraudulent transaction records. To better
+understand this process, let's begin by looking at the schema for the
+credit card transaction data.
+
+....
+CREATE TABLE transactions
+(
+Time VARCHAR,
+V1 VARCHAR,
+V2 VARCHAR,
+V3 VARCHAR,
+V4 VARCHAR,
+V5 VARCHAR,
+V6 VARCHAR,
+V7 VARCHAR,
+V8 VARCHAR,
+V9 VARCHAR,
+V10 VARCHAR,
+V11 VARCHAR,
+V12 VARCHAR,
+V13 VARCHAR,
+V14 VARCHAR,
+V15 VARCHAR,
+V16 VARCHAR,
+V17 VARCHAR,
+V18 VARCHAR,
+V19 VARCHAR,
+V20 VARCHAR,
+V21 VARCHAR,
+V22 VARCHAR,
+V23 VARCHAR,
+V24 VARCHAR,
+V25 VARCHAR,
+V26 VARCHAR,
+V27 VARCHAR,
+V28 VARCHAR,
+Amount VARCHAR,
+Class VARCHAR
+);
+....
+
+Each record in this set consists of a large amount of data that is mostly
+hidden in the
+https://en.wikipedia.org/wiki/Principal_component_analysis[principal components]
+V1 through V28. Due to confidentiality constraints, none of the original
+background information is available about those fields. The fields which have
+not been transformed with
+https://en.wikipedia.org/wiki/Principal_component_analysis[PCA] are `Time` and
+`Amount`. `Time` is the amount of time in seconds that has elapsed between
+each transaction and the first transaction in the set. `Amount` is the amount
+of the transaction. `Class` is a variable that has been added by the data
+maintainers to reflect fraud in the transaction. This set was originally
+conceived to aid in creating machine learning models and as we are using it
+with a different purpose we will be ignoring the `Class` column and also using
+the principal components in a slightly different manner.
+
+To select the fraudulent transactions in this tutorial we will use a Spark in two
+different manners: to select records for inspection, and to distribute business
+rules for processing.
+Initially, Spark will connect to the PostgreSQL database and select all records
+with an `Amount` greater than set value. This is to represent how a database
+query might be run on all transactions above a predefined limit for routine
+fraud inspection. One of the primary reason to use Spark for this selection
+instead of a simple database query is that Spark will transform the results
+from the query into a convenient form for parallel processing.
+
+With the initial set of records selected, Spark will distribute the data to
+be processed in parallel by the business rules. As each record is distributed
+to the processing cluster, the rules will be applied by Drools. For this
+tutorial the rule we are using simply looks at the principal component `V1`
+and marks a transaction as fraudulent if the value is below a predefined
```

Comment:
Is this really "fraudulent if the value is below a...", rather than 'above'?

Reply:
in this case yes, because our simple rule just looks to see if the value that comes from the principle component is below a certain threshold. should i try to state this more explicitly?

Reply:
No that's fine - just a lack of understanding on my part.